packages/plugins/response-cache/src/plugin.ts

```diff
@@ -351,12 +363,15 @@ export function useResponseCache<PluginContext extends Record<string, any> = {}>
             idFieldByTypeName.set(typeName, fieldName);
           }
 
-          if (cacheControlDirective) {
-            const cacheControlAnnotations = getDirective(schema, fieldConfig, 'cacheControl');
+          if (directive) {
+            const cacheControlAnnotations = getDirective(
+              schema,
+              fieldConfig,
+              'cacheControl',
+            ) as unknown as CacheControlDirective[] | undefined;
```

Comment:
The idea of having a `getCacheControlDirective` was to hide the noise of the type casting from the actual which is using it.

I think it could make the code clearer and avoid the duplication since this is done twice :-)

Reply:
On the flip side, then we have an extra function which is an abstractional cost, and all it does is type casting. I believe that this is in a gray zone but imo the layer of indirection and unnecessary jump / context switch while reading code does not justify it keeping things too "DRY".